presto-orc/src/main/java/com/facebook/presto/orc/StreamLayout.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.orc.metadata.Stream;
+import com.facebook.presto.orc.stream.StreamDataOutput;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * StreamLayout is used by the Writer to reorder the data streams.
+ * In ORC, a column is stored in multiple streams (for example
+ * DATA, PRESENT, LENGTH, DICTIONARY and so on).
+ */
+public interface StreamLayout
+{
+    void reOrder(List<StreamDataOutput> dataStreams);
+
+    /**
+     * Streams are ordered by the Stream Size. There is no ordering between
+     * two streams of the same size. This strategy places all small streams
```

Comment:
"This strategy places all small streams" - it orders them by the ascending order of the stream size